src/LearnMe.Core/Services/Calendar/GoogleCalendar.cs

```diff
@@ -0,0 +1,159 @@
+        private readonly CalendarService _calendarService;
+        private readonly string _applicationName = "Learn Me WEB Applicaton";
+        private readonly ICrudRepository<CalendarEvent> _repository;
+        private readonly IMapper _mapper;
+        private readonly ILogger<GoogleCalendar> _logger;
+
+        public GoogleCalendar(IGoogleAPIconnection googleAPIconnection,
+                              ICrudRepository<CalendarEvent> repository,
+                              IMapper mapper,
+                              ILogger<GoogleCalendar> logger)
+        {
+            var token = googleAPIconnection.GetToken();
+            _calendarService = googleAPIconnection.CreateCalendarService(token, _applicationName);
+            _repository = repository;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        //TODO Add Event builder implementation
+        public async Task<bool> CreateEventAsync(CalendarEventDto eventData,
+                                                 string calendarId = "primary",
+                                                 string timezone = "Europe/Warsaw",
+                                                 bool isRecurringEvent = false)
+        {
+            CalendarEvent newDbEvent = _mapper.Map<CalendarEvent>(eventData);
+
+            Event newCalendarEvent = new Event()
+            {
+                Summary = newDbEvent.Title,
+                Description = newDbEvent.Description
+            };
+
+            DateTime? startDateTime = newDbEvent.Start;
+            EventDateTime start = new EventDateTime()
+            {
+                DateTime = startDateTime,
+                TimeZone = timezone
+            };
+            newCalendarEvent.Start = start;
+
+            DateTime? endDateTime = newDbEvent.End;
+            EventDateTime end = new EventDateTime()
+            {
+                DateTime = endDateTime,
+                TimeZone = timezone
+            };
+            newCalendarEvent.End = end;
+
+            if (isRecurringEvent)
+            {
+                String[] recurrence = new String[] { "RRULE:FREQ=DAILY;COUNT=7" };
+                newCalendarEvent.Recurrence = recurrence.ToList();
+            }
+
+            // FOR FUTURE - ADD ATTENDEES
+            //EventAttendee[] attendees = new EventAttendee[]
+            //{
+            //    new EventAttendee() { Email = "<---email-address--->@gmail.com" },
+            //};
+            //newEvent.Attendees = attendees.ToList();
+
+            var createdEvent = await _calendarService.Events.Insert(newCalendarEvent, calendarId).ExecuteAsync();
+            newDbEvent.CalendarId = createdEvent.Id;
+
+            return await _repository.InsertAsync(newDbEvent);
+        }
+
+        public async Task<bool> DeleteEventAsync(int id)
+        {
+            return await _repository.DeleteAsync(id);
+        }
+
+        private void SynchronizeDatabaseWithCalendar(string calendarId = "primary")
```

Comment:
`Primary` to magiczna wartość, jakbyś miała skomplikowaną stałą to masz większe prawdopodobieństwo że ktoś zrobi błąd. Do klasy ze stałymi to